transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -1001,27 +997,24 @@ jint netty_unix_socket_JNI_OnLoad(JNIEnv* env, const char* packagePrefix) {
         netty_unix_errors_throwRuntimeException(env, "failed to get method ID: InetSocketAddress.<init>(String, int)");
         return JNI_ERR;
     }
-    nettyClassName = netty_unix_util_prepend(packagePrefix, "io/netty/util/NetUtil");
-    jclass localNetUtilClass = (*env)->FindClass(env, nettyClassName);
-    free(nettyClassName);
-    nettyClassName = NULL;
-    if (localNetUtilClass == NULL) {
+
+    // We retrieve java.net.preferIPv4Stack directly via Boolean.getBoolean(...) and not via NetUtil to ensure
+    // we not run into deadlocks while trying to load classes that may depend on each other.
+    // See https://github.com/netty/netty/issues/7458
+    jclass booleanClass = (*env)->FindClass(env, "java/lang/Boolean");
+    if (booleanClass == NULL) {
         // pending exception...
         return JNI_ERR;
     }
-    netUtilClass = (jclass) (*env)->NewGlobalRef(env, localNetUtilClass);
-    if (netUtilClass == NULL) {
-        // out-of-memory!
-        netty_unix_errors_throwOutOfMemoryError(env);
-        return JNI_ERR;
-    }
-    netUtilClassIpv4PreferredMethodId = (*env)->GetStaticMethodID(env, netUtilClass, "isIpV4StackPreferred", "()Z" );
-    if (netUtilClassIpv4PreferredMethodId == NULL) {
-        // position method was not found.. something is wrong so bail out
-        netty_unix_errors_throwRuntimeException(env, "failed to get method ID: NetUild.isIpV4StackPreferred()");
+
+    jmethodID booleanClassGetBooleanMethodId = (*env)->GetStaticMethodID(env, booleanClass, "getBoolean", "(Ljava/lang/String;)Z" );
+    if (booleanClassGetBooleanMethodId == NULL) {
+        netty_unix_errors_throwRuntimeException(env, "failed to get method ID: Boolean.getBoolean(String)");
         return JNI_ERR;
     }
 
+    jboolean preferIPv4Stack = (*env)->CallStaticBooleanMethod(env, booleanClass, booleanClassGetBooleanMethodId, "java.net.preferIPv4Stack");
```

Comment:
what happens if a security manager prevents this call from being made?

Reply:
no idea ... @jasontedor any idea ?

That said its the same that the JDK does

Reply:
Maybe check if there is a pending exception after this call and return early?